.github/workflows/build.yml

```diff
@@ -0,0 +1,72 @@
+name: Java CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+      - name: Build Distribution for Launcher
+        run: ./gradlew distForLauncher
+      - name: Unit Tests
+        run: ./gradlew unitTest
+      - name: API Documentation
+        run: ./gradlew javadoc
+      - name: Publish
+        run: echo "TODO"
+  integration-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2      
+      - name: Integration Tests
+        run: ./gradlew integrationTest
+  code-analysis:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2      
+      - name: Static Code Analysis
+        run: ./gradlew check -x test
```

Comment:
would it make sense to split into mandatory (checkstyle) and not mandatory (spotbugs, pmd) ones here?

Reply:
We could - depends a bit on what we want to see in Github and/or how we view the build results.

Currently, we're more interested in the binary :x: / :heavy_check_mark:  than anything else